app/[locale]/institute/sections/page.tsx

```diff
@@ -1,9 +1,91 @@
-import { WorkInProgressStatus } from '~/components/status';
+import Link from 'next/link';
+import { Fragment } from 'react';
+import { IconType } from 'react-icons';
+import {
+  MdLibraryBooks,
+  MdOutlineAccountBalance,
+  MdOutlineAdminPanelSettings,
+  MdOutlineElectricalServices,
+  MdOutlineGroupWork,
+  MdOutlineHealthAndSafety,
+  MdOutlineRealEstateAgent,
+  MdOutlineSecurity,
+  MdOutlineSportsTennis,
+  MdOutlineStore,
+} from 'react-icons/md';
+import { BsHddNetworkFill } from 'react-icons/bs';
 
-export default function Sections({
+import Heading from '~/components/heading';
+import { cn } from '~/lib/utils';
+import { db } from '~/server/db';
+import { getTranslations } from '~/i18n/translations';
+
+export default async function Sections({
   params: { locale },
 }: {
   params: { locale: string };
 }) {
-  return <WorkInProgressStatus locale={locale} />;
+  const sections = await db.query.sections.findMany();
+
+  const SectionIcons: Record<string, IconType> = {
+    accounts: MdOutlineAccountBalance,
+    'central-library': MdLibraryBooks,
+    'central-workshop': MdOutlineGroupWork,
+    'centre-of-computing-and-networking': BsHddNetworkFill,
+    'electrical-maintenance': MdOutlineElectricalServices,
+    estate: MdOutlineRealEstateAgent,
+    'general-administration': MdOutlineAdminPanelSettings,
+    'health-centre': MdOutlineHealthAndSafety,
+    security: MdOutlineSecurity,
+    sports: MdOutlineSportsTennis,
+    store: MdOutlineStore,
+  };
+  const text = (await getTranslations(locale)).Section;
+
+  return (
+    <main className="container">
+      <Heading
+        glyphDirection={'dual'}
+        heading={'h1'}
+        text={text.title.toUpperCase()}
+      />
+      <Fragment>
+        <ul
+          className={cn(
+            'container mb-2 sm:mb-4 lg:mb-6 xl:mb-8',
+            'grid grid-cols-1 gap-2',
+            'sm:grid-cols-2 sm:gap-4',
+            'lg:grid-cols-3 lg:gap-6',
+            'xl:grid-cols-4 xl:gap-8'
+          )}
+        >
+          {sections.map((section, i) => {
+            const IconComponent = SectionIcons[section.urlName];
```

Comment:
Kind of a small nit, but just `Icon` is fine
```suggestion
            const Icon = SectionIcons[section.urlName];
```